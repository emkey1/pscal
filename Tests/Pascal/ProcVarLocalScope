program ProcVarLocalScope;

type
  TProc = procedure(n: integer);
  TFunc = function(x: integer): integer;

procedure IncVar(n: integer);
begin
  writeln('inc ', n);
end;

function AddOffset(x: integer): integer;
begin
  AddOffset := x + 7;
end;

procedure Run;
var
  worker: TProc;
  evaluator: TFunc;
  value: integer;

begin
  value := 2;
  worker := @IncVar;
  evaluator := @AddOffset;
  worker(value);
  worker(value + 1);
  writeln('eval=', evaluator(5));
end;

begin
  Run;
end.
