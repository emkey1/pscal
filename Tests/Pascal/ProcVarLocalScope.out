inc 2
inc 3
eval=12
